src/orc.jl

```diff
@@ -0,0 +1,151 @@
+    else
+        API.LLVMOrcAddEagerlyCompiledIR(orc, r_mod, mod, resolver, ctx)
+    end
+    OrcModule(r_mod[])
+end
+
+function remove!(orc::OrcJIT, mod::OrcModule)
+    LLVM.API.LLVMOrcRemoveModule(orc, mod)
+end
+
+function add!(orc::OrcJIT, obj::MemoryBuffer, resolver = C_NULL, ctx = C_NULL)
+    r_mod = Ref{API.LLVMOrcModuleHandle}()
+    API.LLVMOrcAddObjectFile(orc, r_mod, obj, resolver, ctx)
+    return OrcModule(r_mod[])
+end
+
+function mangle(orc::OrcJIT, name)
+    r_symbol = Ref{Cstring}()
+    LLVM.API.LLVMOrcGetMangledSymbol(orc, r_symbol, name)
+    symbol = unsafe_string(r_symbol[])
+    LLVM.API.LLVMOrcDisposeMangledSymbol(r_symbol[])
+    return symbol
+end
+
+struct OrcTargetAddress
+    ptr::API.LLVMOrcTargetAddress
+end
+Base.cconvert(::Type{API.LLVMOrcTargetAddress}, addr::OrcTargetAddress) = addr
+Base.unsafe_convert(::Type{API.LLVMOrcTargetAddress}, addr::OrcTargetAddress) = addr.ptr
+
+Base.pointer(addr::OrcTargetAddress) = reinterpret(Ptr{Cvoid}, addr.ptr % UInt) # LLVMOrcTargetAddress is UInt64 even on 32-bit
+
+OrcTargetAddress(ptr::Ptr{Cvoid}) = OrcTargetAddress(reinterpret(UInt, ptr))
+
+function create_stub!(orc::OrcJIT, name, initial)
+    LLVM.API.LLVMOrcCreateIndirectStub(orc, name, initial)
+end
+
+function set_stub!(orc::OrcJIT, name, new)
+    LLVM.API.LLVMOrcSetIndirectStubPointer(orc, name, new)
+end
+
+function address(orc::OrcJIT, name)
+    r_address = Ref{API.LLVMOrcTargetAddress}()
+    API.LLVMOrcGetSymbolAddress(orc, r_address, name)
+    OrcTargetAddress(r_address[])
+end
+
+function addressin(orc::OrcJIT, mod::OrcModule, name)
+    r_address = Ref{API.LLVMOrcTargetAddress}()
+    API.LLVMOrcGetSymbolAddressIn(orc, r_address, mod, name)
+    OrcTargetAddress(r_address[])
+end
+
+function callback!(orc::OrcJIT, callback, ctx)
+    r_address = Ref{API.LLVMOrcTargetAddress}()
+    LLVM.API.LLVMOrcLazyCompileCallback(orc, r_address, callback, ctx)
+    return OrcTargetAddress(r_address[])
+end
+
+@checked struct JITEventListener 
+    ref::API.LLVMJITEventListenerRef
+end
+Base.unsafe_convert(::Type{API.LLVMJITEventListenerRef}, listener::JITEventListener) = listener.ref
+
+function register!(orc::OrcJIT, listener::JITEventListener)
+    LLVM.API.LLVMOrcRegisterJITEventListener(orc, listener)
+end
+
+function unregister!(orc::OrcJIT, listener::JITEventListener)
+    LLVM.API.LLVMOrcUnregisterJITEventListener(orc, listener)
+end
+
+GDBRegistrationListener()  = JITEventListener(LLVM.API.LLVMCreateGDBRegistrationListener())
+IntelJITEventListener()    = JITEventListener(LLVM.API.LLVMCreateIntelJITEventListener())
+OProfileJITEventListener() = JITEventListener(LLVM.API.LLVMCreateOProfileJITEventListener())
+PerfJITEventListener()     = JITEventListener(LLVM.API.LLVMCreatePerfJITEventListener())
+
+function JITTargetMachine(;triple = LLVM.triple(),
+                           cpu = "", features = "",
+                           jl_opt_level = Base.JLOptions().opt_level)
```

Comment:
Why **jl_**opt_level? If this JIT is Julia specific, rename it JuliaJITTargetMachine or w/e?